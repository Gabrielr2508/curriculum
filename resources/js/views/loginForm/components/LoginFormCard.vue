<template>
	<v-flex xs12 sm6>
		<v-card>
			<v-layout align-center justify-center>
				<v-flex xs8 offset-xs-2>
					<v-img
						class="mt-4"
						src="../img/logo.png"
						contain
						width="100%"
						position="center"
					/>
				</v-flex>
			</v-layout>

			<v-card-text>
				<v-form
					v-model="valid"
					ref="loginForm"
				>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.email"
									:rules="rules.email"
									label="E-mail*"
									outline
									prepend-icon="email"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.password"
									:rules="rules.password"
									:type="showPassword ? 'text' : 'password'"
									:append-icon="showPassword ? 'visibility' : 'visibility_off'"
									prepend-icon="vpn_key"
									label="Senha*"
									outline
									@click:append="showPassword = !showPassword"
								/>
							</v-flex>
							<v-flex xs12>
								<span class="red--text caption">* campos obrigat√≥rios</span>
							</v-flex>
							<vue-recaptcha
								:sitekey="sitekey"
								ref="invisibleRecaptcha"
								size="invisible"
								badge="bottomleft"
								@verify="onVerifyRecaptcha"
								@expired="resetRecaptcha"
							/>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-container>
					<v-layout justify-space-between>
						<v-btn
							@click="handleForgotPassword"
							dark
							flat
							color="primary"
						>
							Resetar senha
						</v-btn>
						<v-btn
							@click="handleSubmit"
							color="primary"
							:disabled="disableSubmit"
						>
							<v-progress-circular
								v-if="disableSubmit"
								color="primary"
								indeterminate
							/>
							<span v-else>Login</span>
						</v-btn>
					</v-layout>
				</v-container>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
import VueRecaptcha from 'vue-recaptcha';
import validators from '../../../common/validators';
import { mapMutations } from 'vuex';

export default {
	components: {
		VueRecaptcha,
	},
	data() {
		return {
			showPassword: false,
			sitekey: process.env.MIX_SITE_KEY,
			valid: true,
			disableSubmit: false,
			formData: {
				email: '',
				password: '',
				recaptcha_response: null,
			},
			rules: {
				...validators,
			}
		}
	},

	methods: {
		...mapMutations('auth', ['setAuthenticated']),

		onVerifyRecaptcha(response) {
			this.formData.recaptcha_response = response;

			this.resetRecaptcha();

			axios.post(
				'/auth',
				this.formData,
			)
			.then(() => this.setAuthenticated(true))
			.catch((error) => {
				console.log(error.response);
			})
			.finally(() => {
				this.disableSubmit = false;
			});
		},

		resetRecaptcha() {
			this.$refs.invisibleRecaptcha.reset();
		},

		handleSubmit() {
			if (!this.disableSubmit && this.$refs.loginForm.validate()) {
				this.disableSubmit = true;
				this.$refs.invisibleRecaptcha.execute();
			}
		},

		handleForgotPassword() {
			this.$emit('toggleResetForm');
		}
	}
}
</script>

<style>

</style>
