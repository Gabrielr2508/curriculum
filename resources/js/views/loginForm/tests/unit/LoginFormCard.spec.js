import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

import LoginFormCard from '@/views/loginForm/components/LoginFormCard';
import VueRecaptcha from 'vue-recaptcha';

Vue.use(Vuetify);

describe('LoginFormCard', () => {
    const build = () => {
        const wrapper = shallowMount(LoginFormCard);

        return {
            wrapper,
            vueRecaptcha: () => wrapper.find(VueRecaptcha),
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build();

        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders main child components', () => {
        // arrange
        const { vueRecaptcha } = build();

        // assert
        expect(vueRecaptcha().exists()).toBe(true);
    });

    it('passes a binded sitekey prop to vue-recaptcha component', () => {
        // arrange
        const { wrapper, vueRecaptcha } = build();

        wrapper.setData({
            sitekey: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
        });

        // assert
        expect(vueRecaptcha().vm.sitekey).toBe(wrapper.vm.sitekey);
    });
});
