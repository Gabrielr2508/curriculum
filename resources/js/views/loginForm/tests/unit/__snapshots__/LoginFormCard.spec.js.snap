// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormCard renders the component 1`] = `
<v-flex-stub tag="div" xs12="" sm6="">
  <v-card-stub tag="div">
    <v-layout-stub tag="div" align-center="" justify-center="">
      <v-flex-stub tag="div" xs8="" offset-xs-2="">
        <v-img-stub width="100%" contain="true" src="../img/logo.png" position="center" transition="fade-transition" class="mt-4"></v-img-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-card-text-stub>
      <v-form-stub value="true">
        <v-container-stub tag="div">
          <v-layout-stub tag="div" column="">
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub rules="v => !!v || 'Esse campo é obrigatório',v => {
    const pattern = /^(([^<>()[\\]\\\\.,;:\\s@&quot;]+(\\.[^<>()[\\]\\\\.,;:\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$/;
    return pattern.test(v) || 'O E-mail precisa ser válido';
  }" label="E-mail*" outline="" prepend-icon="email" required="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12="">
              <v-text-field-stub rules="v => !!v || 'Esse campo é obrigatório',v => v &amp;&amp; v.length >= 6 || 'A senha precisa ter pelo menos 6 caracteres'" type="password" append-icon="visibility_off" prepend-icon="vpn_key" label="Senha*" counter="" outline="" value=""></v-text-field-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" xs12=""><span class="red--text caption">* campos obrigatórios</span></v-flex-stub>
            <vue-recaptcha-stub sitekey="6LfuXagUAAAAAHLGJo4BaJKb9lJ_hiBsYmrvXVHC" badge="bottomleft" size="invisible"></vue-recaptcha-stub>
          </v-layout-stub>
        </v-container-stub>
      </v-form-stub>
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-container-stub tag="div">
        <v-layout-stub tag="div" justify-space-between="">
          <v-btn-stub color="primary" activeclass="v-btn--active" tag="button" dark="true" flat="true" type="button">
            Resetar senha
          </v-btn-stub>
          <v-btn-stub color="primary" activeclass="v-btn--active" tag="button" type="button"><span>Login</span></v-btn-stub>
        </v-layout-stub>
      </v-container-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-flex-stub>
`;
